Run.hx

```
@@ -157,10 +181,63 @@ function build(rebuild:Bool) {
 			Sys.setCwd("..");
 		}
 	}
-	// run go compiler
-	if (!FileSystem.exists("go4hx") || rebuild) {
+	// run go part of compiler, self built
+	if ((!FileSystem.exists("go4hx") && !FileSystem.exists("go4hx.exe")) ||  rebuild) {
 		Sys.println("build go part of the compiler");
-		Sys.command("go build .");
+		if (Sys.command("go build .") == 0) {
+			// if self built display os and arch built for
+			Sys.println("System:");
+			Sys.command("go env GOOS");
+			Sys.command("go env GOARCH");
+			return;
+		}
+		// if go build fails try using prebuilt binaries
+	}else{
+		// exec already found no need to copy prebuild binary over
+		return;
+	}
+	// prebuilt binaries
+	var goarch = ""; // arch
+	var goos = ""; // os
+	switch systemName.toLowerCase() {
+		case "linux":
+			goos = systemName.toLowerCase();
+			if (process.stdout.readLine() == "x86_64") {
+				goarch = "amd64";
+			}else{
+				goarch = "arm64";
+			}
+		case "mac":
+			goos = "darwin";
+			// check if silicon or intel
+			process = new Process("uname -m");
+			if (process.exitCode(true) != 0)
```

I think it would be better to throw an error if no compatible architecture was found instead of falling back to  `arm64`

---

Yeah good point okay, I'll do that.